dlpy/layers.py

```diff
@@ -114,17 +115,46 @@ class Layer(object):
     def __init__(self, name=None, config=None, src_layers=None):
         self.name = name
         self.config = config
+        self.depth = None
 
         if src_layers is None:
             self.src_layers = None
         else:
-            self.src_layers = list(src_layers)
+            # to be compatible with network
+            if isinstance(src_layers, collections.Iterable):
```

Comment:
what are the options that src_layers would be. we are checking if it is a list and then try to make it list. list() part might be unnecessary